pkg/controller/storageversionmigrator/resourceversion.go

```diff
@@ -0,0 +1,272 @@
+}
+
+func (rv *ResourceVersionController) updateSVM(logger klog.Logger, oldObj, newObj interface{}) {
+	oldSVM := oldObj.(*svmv1alpha1.StorageVersionMigration)
+	newSVM := newObj.(*svmv1alpha1.StorageVersionMigration)
+	logger.V(4).Info("Updating", "svm", klog.KObj(oldSVM))
+	rv.enqueue(newSVM)
+}
+
+func (rv *ResourceVersionController) enqueue(svm *svmv1alpha1.StorageVersionMigration) {
+	key, err := controller.KeyFunc(svm)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %#v: %w", svm, err))
+		return
+	}
+
+	rv.queue.Add(key)
+}
+
+func (rv *ResourceVersionController) Run(ctx context.Context) {
+	defer utilruntime.HandleCrash()
+	defer rv.queue.ShutDown()
+
+	logger := klog.FromContext(ctx)
+	logger.Info("Starting", "controller", rv.controllerName)
+	defer logger.Info("Shutting down", "controller", rv.controllerName)
+
+	if !cache.WaitForNamedCacheSync(rv.controllerName, ctx.Done(), rv.svmSynced) {
+		return
+	}
+
+	go wait.UntilWithContext(ctx, rv.worker, time.Second)
+
+	<-ctx.Done()
+}
+
+func (rv *ResourceVersionController) worker(ctx context.Context) {
+	for rv.processNext(ctx) {
+	}
+}
+
+func (rv *ResourceVersionController) processNext(ctx context.Context) bool {
+	eKey, quit := rv.queue.Get()
+	if quit {
+		return false
+	}
+	defer rv.queue.Done(eKey)
+
+	key := eKey.(string)
+	err := rv.sync(ctx, key)
+	if err == nil {
+		rv.queue.Forget(key)
+		return true
+	}
+
+	klog.FromContext(ctx).V(2).Info("Error syncing SVM resource, retrying", "svm", key, "err", err)
+	rv.queue.AddRateLimited(key)
+
+	return true
+}
+
+func (rv *ResourceVersionController) sync(ctx context.Context, key string) error {
+	logger := klog.FromContext(ctx)
+	startTime := time.Now()
+
+	// SVM is a cluster scoped resource so we don't care about the namespace
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	svm, err := rv.svmListers.Get(name)
+	if err != nil {
+		return err
+	}
+	// working with copy to avoid race condition between this and migration controller
+	toBeProcessedSVM := svm.DeepCopy()
+	gvr := getGVRFromResource(toBeProcessedSVM)
+
+	exists, err := rv.resourceExists(gvr)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("resource %q does not exist", gvr.String())
+	}
+
+	if toBeProcessedSVM.Status.ResourceVersion != "" {
+		logger.V(4).Info("Resource version is already set", "svm", name)
+		return nil
+	}
+
+	toBeProcessedSVM.Status.ResourceVersion, err = rv.getListResourceVersion(gvr, ctx)
+	if err != nil {
+		return err
+	}
+
+	_, err = updateSVMStatus(
+		ctx,
+		rv.kubeClient,
+		toBeProcessedSVM,
+		rvAddedConditionType,
+		rvStatusReason,
+	)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Resource version has been successfully added", "svm", key, "elapsed", time.Since(startTime))
+	return nil
+}
+
+func (rv *ResourceVersionController) getListResourceVersion(gvr schema.GroupVersionResource, ctx context.Context) (string, error) {
```

Comment:
```suggestion
func (rv *ResourceVersionController) getLatestResourceVersion(gvr schema.GroupVersionResource, ctx context.Context) (string, error) {
```

Just a suggestion. This might help future readers better understand what the intent is here.